.github/workflows/build.yml

```diff
@@ -0,0 +1,91 @@
+name: build
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
+
+permissions:
+  contents: write
+  id-token: write
+  packages: write
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: arduino/setup-task@v1
+        with:
+          version: 3.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ./dist/*.deb
+            ./dist/*.rpm
+            ./dist/*.apk
+          key: ${{ runner.os }}-go-${{ hashFiles('**/*.go') }}-${{ hashFiles('go.sum') }}--${{ hashFiles('.goreleaser.yml') }}
+      - run: task ci
+      - uses: codecov/codecov-action@v2
+        with:
+          file: ./coverage.txt
+      - run: git diff
+      - name: Install GoReleaser
+        if: startsWith(github.ref, 'refs/tags/v') ||  github.ref == 'refs/heads/main'
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          install-only: true
+      - name: goreleaser-release
+        if: startsWith(github.ref, 'refs/tags/v') ||  github.ref == 'refs/heads/main'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_PAT }}
+          FURY_TOKEN: ${{ secrets.FURY_TOKEN }}
+        run: task goreleaser
+  cranlogs-check-pkgs:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
+    needs:
+      - goreleaser
+    if: github.ref == 'refs/heads/main'
```

Comment:
this takes a couple minutes - and it seems goreleasers pattern is to only care about testing on stuff merged to main, then I guess procedurally tag if everything is green. I'm not sure how i feel about this - it might seem better if it also runs a test on tag prior to actually publishing, at least for me usually i'm bumping docs/tagging and pushing the commit + tag at the same time 🤷  this doesn't really protect properly if the unit tests fail even on the tag (which conversely could be a feature if you just want to YOLO publish)